app/config/message-queue.js

```diff
@@ -1,42 +1,44 @@
-const Joi = require('joi')
+import joi from 'joi'
+import appInsights from 'applicationinsights'
+
 const msgTypePrefix = 'uk.gov.ffc.ahwr'
 
 const sharedConfigSchema = {
-  appInsights: Joi.object(),
-  host: Joi.string(),
-  password: Joi.string(),
-  username: Joi.string(),
-  useCredentialChain: Joi.bool().default(false)
+  appInsights: joi.object(),
+  host: joi.string().required(),
+  password: joi.string(),
+  username: joi.string(),
+  useCredentialChain: joi.bool().default(false)
 }
 
-const schema = Joi.object({
+const schema = joi.object({
   applicationPaymentRequestQueue: {
-    address: Joi.string(),
-    type: Joi.string(),
+    address: joi.string(),
+    type: joi.string(),
     ...sharedConfigSchema
   },
   paymentRequestTopic: {
-    address: Joi.string(),
+    address: joi.string(),
     ...sharedConfigSchema
   },
   paymentResponseSubscription: {
-    topic: Joi.string().default('paymentResponseTopic'),
-    address: Joi.string(),
-    type: Joi.string().default('subscription'),
+    topic: joi.string().default('paymentResponseTopic'),
+    address: joi.string(),
+    type: joi.string().default('subscription'),
     ...sharedConfigSchema
   },
-  submitPaymentRequestMsgType: Joi.string()
+  submitPaymentRequestMsgType: joi.string()
 })
 
 const sharedConfig = {
-  appInsights: require('applicationinsights'),
+  appInsights,
   host: process.env.MESSAGE_QUEUE_HOST,
   password: process.env.MESSAGE_QUEUE_PASSWORD,
   username: process.env.MESSAGE_QUEUE_USER,
   useCredentialChain: process.env.NODE_ENV === 'production'
 }
 
```

Comment:
I'm wondering if this achieves what we want it to achieve. When you import the final product config exported from this file, does it definitely run the code 'down the page'? Does it definitely run it against the schema?

Poor timing to ask this question I know, as we have this pattern in a few repos. If we arent sure or we think this is an issue, it might be better done like this:

```
const buildConfig = () => {
    const config = { ...stuff }
    const schema = joi.schema()
    const { error } = schema.validate(config, { abortEarly: false })

    if (error) {
      throw new Error(`The message queue config is invalid. ${error.message}`)
    }

  return config
}

export config = buildConfig()
```

Reply:
confirmed this is ok, but in the future we might want to move to a function like the above